### ISSUE TITLE
Couldn't build with rustc-1.4.0-nightly

### ISSUE THREAD

**Issue by vanzef** [thepowersgang/rust-barebones-kernel#8]
I've tried to build kernel like in your `README`, but with some differences.

Firstly, I've tried to do `make`, but it fails with this error:

```
rustc -O --cfg arch__amd64 --target=arch/amd64/target.json --cfg disable_float -C soft-float -o .obj/amd64/libcore.rlib --crate-type=lib --emit=link,dep-info ../libcore/lib.rs
warning: ignoring specified output filename because multiple outputs were requested
LLVM ERROR: SSE register return with SSE disabled
Makefile:61: recipe for target '.obj/amd64/libcore.rlib' failed
make: *** [.obj/amd64/libcore.rlib] Error 1
```

So, I decided to apply patch from the root of repo (`libcore_nofp.patch`). It also fails:

```
patching file clone.rs
patching file default.rs
patching file fmt/mod.rs
patching file intrinsics.rs
patching file lib.rs
patching file num/flt2dec/decoder.rs
Hunk #1 succeeded at 12 with fuzz 2.
patching file num/mod.rs
Hunk #1 succeeded at 94 (offset 1 line).
Hunk #2 FAILED at 1403.
1 out of 2 hunks FAILED -- saving rejects to file num/mod.rs.rej
patching file ops.rs
```

And if I built it with wrong-applied patch it says me about 12 errors like `../libcore/num/dec2flt/rawfp.rs:144:1: 189:2 error: the trait`num::Float`is not implemented for the type`f32`[E0277]`

I am using latest (today) nightly:
    rustc --version
    rustc 1.4.0-nightly (e2bebf32f 2015-08-14)

And libcore from `rust-lang/rust` repo.

Could you fix the patch, please?

**Comment by thepowersgang** [comment]
Thanks for mentioning that - This was broken by the VERY recent changes to floating point parsing.

**Comment by thepowersgang** [comment]
Fixed in c4583b63b0cd2ffcd6c0ac1f5169505766d2467e

**Comment by vanzef** [comment]
Thanks!
